src/lib/message_filters.py

```diff
@@ -30,3 +31,22 @@ def filter_time_range(messages, time_key, start_time, end_time):
     @staticmethod
     def filter_noise(messages, message_key, noise_fn):
         return [td for td in messages if not noise_fn(td.get(message_key))]
+    
+    @staticmethod
+    def subsample_messages(messages):
+        '''
+        Generates sample messages for labelling in coda
+        
+        :param: messages: TracedData objects to sample
+        :type traced_data: TracedData
+        :return: sample of the TracedData objects
+        :rtype: TracedData
+        '''
+
+        subsample_data = []
+        for td in messages:
+            if (td["uid"] not in subsample_data) and (int(SHAUtils.sha_string(td["uid"])[0], 16) < 4):
+                subsample_data.append(td)
+        log.info(f"Number of sample messages "
```

Comment:
Probably clearer to say something like " / n *total* messages"

Reply:
Updated thanks!